zebROS_ws/src/ros_control_boilerplate/src/frc_robot_interface.cpp

```diff
@@ -2570,6 +2585,139 @@ void FRCRobotInterface::write(const ros::Time& time, const ros::Duration& period
 		}
 	}
 
+	this->write_tracer_.start_unique("candle");
+	for (size_t candle_id = 0; candle_id < this->num_candles_; candle_id++) {
+		//ROS_INFO_STREAM("Candle_status_0");
+		if (!(this->candle_local_hardwares_[candle_id]))
+			continue;
+		
+		// Get candle, state, and command interfaces
+		auto &candle = this->candles_[candle_id];
+		auto &candle_state = this->candle_state_[candle_id];
+		auto &candle_command = this->candle_command_[candle_id];
+
+		// For each Command property, check if it changed
+		//  if it did change, update the actual CANdle, and update the state
+
+		// Brightness
+		//ROS_INFO_STREAM("Candle_status_1");
+		double brightness;
+		if (candle_command.brightnessChanged(brightness)) {
+			if (safeTalonCall(
+				candle->ConfigBrightnessScalar(brightness),
+				"candle->ConfigBrightnessScalar",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set brightness to " << brightness);
+				candle_state.setBrightness(brightness);
+			} else {
+				candle_command.resetBrightnessChanged();
+			}
+		}
+
+		// Show status LED when active
+		//ROS_INFO_STREAM("Candle_status_2");
+		bool status_led_when_active;
+		if (candle_command.statusLEDWhenActiveChanged(status_led_when_active)) {
+			if (safeTalonCall(
+				candle->ConfigStatusLedState(!status_led_when_active),
+				"candle->ConfigStatusLedState",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set status led when active to " << status_led_when_active);
+				candle_state.setStatusLEDWhenActive(status_led_when_active);
+			} else {
+				candle_command.resetStatusLEDWhenActiveChanged();
+			}
+		}
+
+		// Enabled
+		//ROS_INFO_STREAM("Candle_status_3");
+		bool enabled;
+		if (candle_command.enabledChanged(enabled)) {
+			if (safeTalonCall(
+				candle->configV5Enabled(enabled),
+				"candle->configV5Enabled",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set enabled to " << enabled);
+				candle_state.setEnabled(enabled);
+			} else {
+				candle_command.resetEnabledChanged();
+			}
+		}
+
+		// Animation
+		//ROS_INFO_STREAM("Candle_status_4");
+		CANdleAnimation candle_animation;
+		Animation* animation;
+		if (candle_command.animationChanged(candle_animation)) {
+			ROS_INFO_STREAM("Candle_status_4.1");
+			// Convert from CANdleAnimation to the appropriate CTRE animation class
+			if (candle_animation.class_type == CANdleAnimationClass::BaseStandard) {
+				ROS_INFO_STREAM("Candle_status_4.1.1");
```

Comment:
And definitely remove these non-printed out debugging bits
Keep any actual status printouts when things change, though - those are nice for confirming things work, just as long as they're not printing every loop iteration